packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -231,6 +269,25 @@ export class CfnInclude extends core.CfnElement {
     this.conditions[conditionName] = cfnCondition;
   }
 
+  private createNestedStack(nestedStackId: string, props?: CfnIncludeProps): void {
+    if (nestedStackId in this.template.Resources) {
+      if (this.template.Resources[nestedStackId].Type === 'AWS::CloudFormation::Stack' && props?.nestedStacks) {
+        // add 'Nested' to logicalId so that we don't conflict with the 'AWS::CloudFormation::Stack' resource.
```

Comment:
Of course, this is where the problem is.

You're right that you shouldn't conflict, but this is not how to solve this problem :).

You should:

1. If the nested stack with ID `nestedStackId` has been provided in `nestedStacks` when creating `CfnInclude`, you should _not_ create the `CfnStack` resource for it.
2. Instead, the `CfnStack` will be created by the `NestedStack` class.
3. You need to override the `logicalId` property of the `CfnStack` created inside `NestedStack` (it's available as the `nestedStackResource` property on it), similarly like we do for regular resources.